src/textwatcher.js

```diff
@@ -133,7 +133,10 @@ export default class TextWatcher {
 		this.hasMatch = textHasMatch;
 
 		if ( textHasMatch ) {
-			const eventData = Object.assign( data, { text, range } );
+			// If text matches, get the normalized transformation from the match and then pass it to the
+			// event data object.
+			const normalizedTransformation = textHasMatch.normalizedTransformation;
```

Comment:
I didn't notice this before, but there's a problem here. \`TextWatcher\` is all about allowing you to react to some text changes. It's not said how you want to react. So, it does not have anything to do with the text transformations. Hence – any changes in this class should be kept generic.

Sorry for that, but you need to change this to something like `textHasMatch.additionalData`.